File: SPSP/SPSP.Models/Report/CustomerReportData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SPSP.Models
{
    public class CustomerReportData
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }

        public int OrderCount { get; set; }

        public decimal TotalAmount { get; set; }    
    }
}

File: SPSP/UI/spsp_desktop/lib/models/customer_report_data/customer_report_data.dart
import 'package:json_annotation/json_annotation.dart';

part 'customer_report_data.g.dart';

@JsonSerializable()
class CustomerReportData {
  int? id;
  String? firstName;
  String? lastName;
  String? phone;
  String? email;
  int orderCount;
  double totalAmount;

  CustomerReportData({
    this.id,
    this.firstName,
    this.lastName,
    this.phone,
    this.email,
    required this.orderCount,
    required this.totalAmount,
  });

  factory CustomerReportData.fromJson(Map<String, dynamic> json) =>
      _$CustomerReportDataFromJson(json);

  Map<String, dynamic> toJson() => _$CustomerReportDataToJson(this);
}
